api/routes.go

```diff
@@ -85,15 +92,29 @@ func (r *RouteHandler) recommendClusterSetup(c *gin.Context) {
 	log.Info("recommend cluster setup")
 	provider := c.Param("provider")
 	region := c.Param("region")
-	var request recommender.ClusterRecommendationReq
-	if err := c.BindJSON(&request); err != nil {
-		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+
+	// request decorated with provider and region
+	reqWr := RequestWrapper{P: provider, R: region}
+
+	if err := c.BindJSON(&reqWr); err != nil {
+		c.JSON(http.StatusBadRequest, map[string]string{
```

Comment:
log an error or warning

Reply:
Use `gin.H{}` instead of `map[string]string`

Reply:
fixed